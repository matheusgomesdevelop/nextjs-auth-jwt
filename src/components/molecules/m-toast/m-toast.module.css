.m_toast__content {
  display: flex;
  gap: var(--spacing-16);
}
.m_toast__message {
  padding: var(--spacing-24) var(--spacing-16);
}

.m_toast__icon {
}

.m_toast__text {
  display: flex;
  flex-direction: column;
  gap: var(--spacing-6);
}
