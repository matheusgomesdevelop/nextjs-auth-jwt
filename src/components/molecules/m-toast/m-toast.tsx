import { Toast, ToastMessage, ToastProps } from "primereact/toast";
import { createRef, forwardRef, useImperativeHandle } from "react";

// StyleSheet
import styles from "./m-toast.module.css";

export interface I_MToastMessage extends ToastMessage {}

interface I_MToast extends ToastProps {
  "data-testid"?: string;
  message: I_MToastMessage | I_MToastMessage[];
}

interface I_MToastHandle {
  showToast: () => void;
}

export interface I_MToastComponent extends Toast {
  showToast: () => void;
}

export const M_Toast = forwardRef<I_MToastHandle, I_MToast>(function M_Toast(props, ref) {
  const mToastRef = createRef<I_MToastComponent>();

  useImperativeHandle(
    ref,
    () => {
      return {
        showToast() {
          if (mToastRef && mToastRef.current) {
            mToastRef.current.show(props.message);
          }
        },
      };
    },
    [mToastRef, props.message]
  );

  return (
    <Toast
      data-testid="m-toast"
      pt={{
        message: {
          className: styles.m_toast__message,
        },
        content: {
          className: styles.m_toast__content,
        },
        icon: {
          className: styles.m_toast__icon,
        },
        text: {
          className: styles.m_toast__text
        }
      }}
      ref={mToastRef}
      {...props}
    />
  );
});
