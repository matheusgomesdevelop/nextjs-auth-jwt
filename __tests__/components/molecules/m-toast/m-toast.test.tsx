import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

// Molecules
import { M_Toast } from "../../../../src/components/molecules/m-toast/m-toast";

describe("Deve renderizar o m-toast, corretamente", () => {
  it("Deve renderizar o componente na tela", () => {
    render(
      <M_Toast
        message={{
          severity: "success",
          summary: "Success",
          detail: "Message Content",
          life: 3000,
        }}
      />
    );

    const get_toast = screen.getByTestId("m-toast");

    expect(get_toast).toBeInTheDocument();
  });
  it("Deve manter a estrutura visual do Toast", () => {
    const m_toast = renderer
      .create(
        <M_Toast
          message={{
            severity: "success",
            summary: "Success",
            detail: "Message Content",
            life: 3000,
          }}
        />
      )
      .toJSON();

    expect(m_toast).toMatchSnapshot();
  });
});
