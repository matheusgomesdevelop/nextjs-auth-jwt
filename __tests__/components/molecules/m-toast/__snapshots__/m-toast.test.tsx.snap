// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o m-toast, corretamente Deve manter a estrutura visual do Toast 1`] = `null`;
